--- xynth-0.8.90.orig/Makefile	2007-09-25 10:39:10.000000000 +0200
+++ xynth-0.8.90/Makefile	2009-02-19 19:40:58.000000000 +0100
@@ -5,12 +5,12 @@
 .PHONY := all clean dist install optimize doxy-doc xynth.pc test strip update pspdev
 
 all:
-	@$(MAKE) --no-print-directory -C src
+	@(cd src && $(MAKE))
 
 clean:
-	@$(MAKE) --no-print-directory clean -C src
-	@$(MAKE) --no-print-directory clean -C tools/theme
-	@$(MAKE) --no-print-directory clean -C tools/optimize
+	@(cd src && $(MAKE) clean)
+	@(cd tools/theme && $(MAKE) clean)
+	@(cd tools/optimize && $(MAKE) clean)
 	rm -f xynth.pc
 	rm -rf dist
 	rm -rf doc
@@ -24,7 +24,7 @@
 	mkdir -p $(DISTCONFDIR)
 	mkdir -p $(DISTFONTDIR)
 	mkdir -p $(DISTTHEMEDIR)
-	@$(MAKE) --no-print-directory dist -C src
+	@(cd src && $(MAKE) dist)
 	mkdir -p $(DISTPKGCONFIG)
 	cp xynth.pc $(DISTPKGCONFIG)/xynth.pc
 
--- xynth-0.8.90.orig/Makefile.cfg	2007-09-25 10:39:10.000000000 +0200
+++ xynth-0.8.90/Makefile.cfg	2009-03-08 19:17:40.000000000 +0100
@@ -1,9 +1,9 @@
 
 # Host Settings
-HOSTCC     = gcc
+#HOSTCC     = gcc
 HOSTCXX    = g++
 HOSTAR     = ar rcs
-HOSTMAKE   = make
+#HOSTMAKE   = make
 HOSTRANLIB = chmod a+r
 HOSTSTRIP  = strip -xsR .note -R .comment
 HOSTNM     = nm
@@ -15,10 +15,10 @@
 PLATFORM_GP2X   = N
 
 # Platform Settings
-CC     = gcc
+#CC     = gcc
 CXX    = g++
 AR     = ar rcs
-MAKE   = make
+#MAKE   = make
 RANLIB = chmod a+r
 STRIP  = strip -xsR .note -R .comment
 NM     = nm
@@ -39,10 +39,10 @@
 # GDI         	: GDI - http://msdn.microsoft.com/
 # GP2X          : Gamepark Holdings GP2X - http://wiki.gp2x.org
 # NULL		: NULL
-VIDEO_VESA      = Y
+VIDEO_VESA      = N
 VIDEO_SVGALib   = N
 VIDEO_FBDev     = Y
-VIDEO_SDL       = Y
+VIDEO_SDL       = N
 VIDEO_PSPDEV    = N
 VIDEO_GDI       = N
 VIDEO_GP2X      = N
@@ -68,7 +68,7 @@
 PIPE_MEM        = N
 
 # Poll Support
-HAVE_POLL       = Y
+HAVE_POLL       = N
 
 # Image Support
 IMAGE_PNG       = Y
@@ -96,11 +96,11 @@
 DEMO_OBJECT     = Y
 DEMO_ONSCREENKEYBOARD = Y
 DEMO_TEMP       = Y
-DEMO_TERM       = Y
+DEMO_TERM       = N
 DEMO_TIMER      = Y
 DEMO_LOGOUT     = Y
 DEMO_IMAGE      = Y
-DEMO_WIDGETX    = Y
+DEMO_WIDGETX    = N
 
 CFLAGS += -Wall -Wparentheses -pipe -fsigned-char -fno-strict-aliasing -DREENTRANT
 
--- xynth-0.8.90.orig/Makefile.rules	2007-09-25 10:39:10.000000000 +0200
+++ xynth-0.8.90/Makefile.rules	2009-02-20 02:10:29.000000000 +0100
@@ -5,18 +5,18 @@
     SECHO := @-false
     DISP  := sil
     Q     := @
-    MAKE  := make --no-print-directory
+#    MAKE  := make --no-print-directory
 else
 ifneq ($(V)$(VERBOSE),)
     SECHO := @-false
     DISP  := ver
     Q     := 
-    MAKE  := make -w
+#    MAKE  := make -w
 else
     SECHO := @echo
     DISP  := pur
     Q     := @
-    MAKE  := make --no-print-directory
+#    MAKE  := make --no-print-directory
 endif
 endif
 
@@ -32,7 +32,7 @@
 SED  = sed
 
 HOSTLD     = ld
-HOSTCC     = gcc
+HOSTCC     = $(CC)
 HOSTCXX    = g++
 HOSTAR     = ar rcs
 HOSTRANLIB = ranlib
