--- gettext-0.14.6.orig/gettext-runtime/intl/loadmsgcat.c	2008-11-02 01:54:51.000000000 +0100
+++ gettext-0.14.6/gettext-runtime/intl/loadmsgcat.c	2008-11-02 01:55:27.000000000 +0100
@@ -1011,7 +1011,7 @@
       /* The magic number is wrong: not a message catalog file.  */
 #ifdef HAVE_MMAP
       if (use_mmap)
-	munmap ((caddr_t) data, size);
+	munmap (data, size);
       else
 #endif
 	free (data);
@@ -1376,7 +1376,7 @@
 	free (domain->malloced);
 #ifdef HAVE_MMAP
       if (use_mmap)
-	munmap ((caddr_t) data, size);
+	munmap (data, size);
       else
 #endif
 	free (data);
@@ -1410,7 +1410,7 @@
 
 # ifdef _POSIX_MAPPED_FILES
   if (domain->use_mmap)
-    munmap ((caddr_t) domain->data, domain->mmap_size);
+    munmap (domain->data, domain->mmap_size);
   else
 # endif	/* _POSIX_MAPPED_FILES */
     free ((void *) domain->data);
